### PR TITLE
Fix webpack for multiple enviroments and rvm issue with EnvironmentPl…

### PR DESCRIPTION
**Pull request by djforth**
Fixes for https://github.com/rails/webpacker/issues/167 & https://github.com/rails/webpacker/issues/158

### CODE REVIEW

File: lib/install/config/shared.js [hunk 1]
```diff
@@ -50,7 +50,7 @@ const config = {
   },
 
   plugins: [
-    new webpack.EnvironmentPlugin(Object.keys(process.env))
+    new webpack.EnvironmentPlugin(Object.keys(process.env).filter((key)=>!key.match(/rvm/))
```
Comment (ytbryan) [inline lib/install/config/shared.js]: A missing closing parenthesis

Comment (djforth) [inline lib/install/config/shared.js]: Sorry my bad, my sublimelinter didn't pick it up. Just pushing fix

**Comment by gauravtiwari** [comment]
@djforth This is not just related to rvm, but with other env keys that are either `null or undefined`. I have wrapped up this in another PR - https://github.com/rails/webpacker/pull/153/files#diff-67f77b829953e34e910db8d7c0c285c3R66 

That will fix the problem once merged 🎉

**Comment by djforth** [comment]
Ah I see ok excellent :)

**Comment by noma4i** [comment]
This PR doesn't fix https://github.com/rails/webpacker/issues/158

**Comment by gauravtiwari** [comment]
@noma4i but this will #153, once merged.

**Pull request closed without merging**